torch/cuda/__init__.py

```diff
@@ -544,6 +547,15 @@ def _device_count_nvml() -> int:
         return raw_cnt
     return len(set(range(raw_cnt)).intersection(visible_devices))
 
+def _get_nvml_device_index(device: Optional[Union[int, Device]]) -> int:
+    r"""Returns the NVML index of the device, taking CUDA_VISIBLE_DEVICES into account."""
+    idx = _get_device_index(device, optional=True)
+    visible_devices = _parse_visible_devices()
```

Comment:
Q: Should this also work for MIG- prefixed devices, as described in https://github.com/pytorch/pytorch/issues/90543 ?

Reply:
This is not working with MIG devices. The device handle is retrieved by NVML index:

```python
handle = pynvml.nvmlDeviceGetHandleByIndex(device)
```

If we want to support MIG devices here, it's better to use UUIDs:

```python
uuid = _get_nvml_device_uuid(device)
handle = pynvml.nvmlDeviceGetHandleByUUID(uuid)
```

then the returned can be a GPU device handle or a MIG device handle.